### PR TITLE
feat(ts): add virtual index parameters

### LINKED ISSUE #1249: feat(virtual indices): Virtual indices related parameters
**Summary**

Virtual indices feature (aka virtual replicas) provides a few new parameters:

Search & Settings parameters:
* **relevancyStrictness** 
  - integer value [0-100]  
  - relevancy score to apply to search in virtual index. A bigger value means fewer, but more relevant results, lesser value means more, but less relevant results

Search response:
* **appliedRelevancyStrictness** 
  - integer value [0-100]
  - relevancy score to apply to the search in virtual index. 
* **nbSortedHits** 
   - integer value
   - number of relevant hits to display in case of non-zero `relevancyStrictness` applied

### PR DESCRIPTION
**Pull request by Haroenv**
fixes #1249

### CODE REVIEW

File: packages/client-search/src/types/Settings.ts [hunk 1]
```diff
@@ -297,4 +297,11 @@ export type Settings = {
    * transliteration to make words indexed in Katakana or Kanji searchable in Hiragana.
    */
   readonly attributesToTransliterate?: readonly string[];
+
+  /**
+   * relevancy score to apply to search in virtual index. A Bigger value means
+   * fewer, but more relevant results, smaller value means more, but less
+   * relevant results.
+   */
```
Comment (Haroenv) [inline packages/client-search/src/types/Settings.ts:305]:
```suggestion
  /**
   * The relevancy score to apply to search in a virtual index [0-100]. A Bigger
   * value means fewer, but more relevant results, smaller value means more, but
   * less relevant results.
   */
```

File: packages/client-search/src/types/SearchResponse.ts [hunk 2]
```diff
@@ -28,6 +28,11 @@ export type SearchResponse<TObject = {}> = {
    */
   nbHits: number;
 
+  /**
+   * Subset of hits matched when relevancyStrictness is applied.
```
Comment (Ant-hem) [inline packages/client-search/src/types/SearchResponse.ts:32]:
```suggestion
   * Subset of hits selected when relevancyStrictness is applied.
```

File: packages/client-search/src/types/SearchResponse.ts [hunk 3]
```diff
@@ -215,4 +220,9 @@ export type SearchResponse<TObject = {}> = {
      */
     params?: Record<string, any>;
   };
+
+  /**
+   * The relevancy score applied to search in a virtual index
```
Comment (Ant-hem) [inline packages/client-search/src/types/SearchResponse.ts:225]:
```suggestion
   * The relevancy threshold applied to search in a virtual index
```

File: packages/client-search/src/types/SearchOptions.ts [hunk 4]
```diff
@@ -342,4 +342,11 @@ export type SearchOptions = {
    * results.
    */
   readonly decompoundQuery?: boolean;
+
+  /**
+   * The relevancy score to apply to search in a virtual index [0-100]. A Bigger
```
Comment (Ant-hem) [inline packages/client-search/src/types/SearchOptions.ts:347]:
```suggestion
   * The relevancy threshold to apply to search in a virtual index [0-100]. A Bigger
```

File: packages/client-search/src/types/Settings.ts [hunk 5]
```diff
@@ -297,4 +297,11 @@ export type Settings = {
    * transliteration to make words indexed in Katakana or Kanji searchable in Hiragana.
    */
   readonly attributesToTransliterate?: readonly string[];
+
+  /**
+   * The relevancy score to apply to search in a virtual index [0-100]. A Bigger
```
Comment (Ant-hem) [inline packages/client-search/src/types/Settings.ts:302]:
```suggestion
   * The relevancy threshold to apply to search in a virtual index [0-100]. A Bigger
```

**Comment by Haroenv** [comment]
thanks for your reviews @Ant-hem & @samouss !

File: packages/client-search/src/types/SearchResponse.ts [hunk 6]
```diff
@@ -28,6 +28,11 @@ export type SearchResponse<TObject = {}> = {
    */
   nbHits: number;
 
+  /**
+   * Subset of hits selected when relevancyStrictness is applied.
+   */
+  nbSortedHits: number;
```
Comment (Ant-hem) [inline packages/client-search/src/types/SearchResponse.ts:34]: Sorry, I didn't see it in my first review but, `nbSortedHits` is optional and will only be present when querying VRs.
Is that good with your current implem?

Comment (Haroenv) [reply packages/client-search/src/types/SearchResponse.ts:34]: ah yes thanks for catching, it's meant to have a question mark!

```suggestion
  nbSortedHits?: number;
```

**Review by Ant-hem — APPROVED** [review]
LGTM

**Pull request merged**